pkg/controller/nodemaintenance/check_lease_supported_test.go

```diff
@@ -0,0 +1,62 @@
+package nodemaintenance
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+
+
+func areErrorsEqual(err1, err2 error) bool {
+    return (err1 != nil && err2 != nil && err1.Error() == err2.Error()) ||
+        (err1 == nil && err2 == nil)
+}
+
+type mockCreator func() *FakeClient
+
+type TestCaseDefinition struct {
+	info        string
+	mock		*FakeClient
+	expectedErr error
+	expectedRes bool
+}
+
+var _ = Describe("checkIfLeaseAPISupported", func() {
+	It("check lease api test cases", func() {
```

Comment:
This was at my request.  I originally asked for something like https://github.com/kubernetes/dashboard/blob/master/src/app/backend/validation/validaterbacstatus_test.go so that we could use the same pattern elsewhere.

Reply:
That's a nice pattern for plain go tests indeed, but IMHO with ginkgo / gomega `DescribeTable` is more convenient. But it's not critical for this PR of course.